install.sh

```diff
@@ -180,6 +181,46 @@ detect_device() {
   fi
 }
 
+# Function to check if a port is available
+check_port() {
+  port=$1
+  if $SUDO lsof -nP -iTCP:"$port" -sTCP:LISTEN >/dev/null 2>&1; then
+    return 1
+  fi
+  return 0
+}
+
+# Function to check if the server and worker ports are available
+check_ports() {
+  config_file=$(get_param_value "config-file" "$@")
```

Comment:
Can we support check ports defined in config-file as well?

Reply:
I would like to consider this an advanced use case. Keep it simple to avoid additional dependencies or handling yaml parsing in shell.